src/Samples/SarifToCsv/Program.cs

```diff
@@ -85,70 +98,109 @@ public static void ConvertSarifLog(JsonSerializer serializer, string sarifFilePa
                         }
                     }
+
+                    context.RunIndex++;
                 }
             }
         }
 
-        public static Dictionary<string, Action<CsvWriter, Result, PhysicalLocation>> BuildWriters()
+        public static void Write(WriteContext context, IEnumerable<Action<WriteContext>> columnWriters)
         {
-            Dictionary<string, Action<CsvWriter, Result, PhysicalLocation>> writers = new Dictionary<string, Action<CsvWriter, Result, PhysicalLocation>>(StringComparer.OrdinalIgnoreCase);
+            foreach (Action<WriteContext> columnWriter in columnWriters)
+            {
+                columnWriter(context);
+            }
 
-            // Result Basic Properties
-            writers["BaselineState"] = (writer, result, pLoc) => { writer.Write(result.BaselineState.ToString()); };
-            writers["CorrelationGuid"] = (writer, result, pLoc) => { writer.Write(result.CorrelationGuid ?? ""); };
-            writers["Fingerprints"] = (writer, result, pLoc) => { writer.Write(String.Join("; ", result.Fingerprints?.Values ?? Array.Empty<string>())); };
-            writers["HostedViewerUri"] = (writer, result, pLoc) => { writer.Write(result.HostedViewerUri?.ToString() ?? ""); };
-            writers["InstanceGuid"] = (writer, result, pLoc) => { writer.Write(result.InstanceGuid ?? ""); };
-            writers["Level"] = (writer, result, pLoc) => { writer.Write(result.Level.ToString()); };
-            writers["Message.Text"] = (writer, result, pLoc) => { writer.Write(result.Message?.Text ?? ""); };
-            writers["OccurrenceCount"] = (writer, result, pLoc) => { writer.Write(result.OccurrenceCount); };
-            writers["PartialFingerprints"] = (writer, result, pLoc) => { writer.Write(String.Join("; ", result.PartialFingerprints?.Values ?? Array.Empty<string>())); };
-            writers["Provenance"] = (writer, result, pLoc) => { writer.Write(result.Provenance?.ToString() ?? ""); };
-            writers["Rank"] = (writer, result, pLoc) => { writer.Write(result.Rank.ToString()); };
-            writers["RuleId"] = (writer, result, pLoc) => { writer.Write(result.RuleId ?? ""); };
-            writers["RuleIndex"] = (writer, result, pLoc) => { writer.Write(result.RuleIndex); };
-            writers["SuppressionStates"] = (writer, result, pLoc) => { writer.Write(result.SuppressionStates.ToString()); };
-            writers["Tags"] = (writer, result, pLoc) => { writer.Write(String.Join("; ", ((IEnumerable<string>)result.Tags) ?? Array.Empty<string>())); };
-            writers["WorkItemUris"] = (writer, result, pLoc) => { writer.Write(String.Join("; ", result.WorkItemUris?.Select((uri) => uri.ToString()) ?? Array.Empty<string>())); };
+            context.Writer.NextRow();
+            if(context.Writer.RowCountWritten % 1000 == 0) { Console.Write("."); }
+        }
+
+        public static Dictionary<string, Action<WriteContext>> BuildWriters()
```

Comment:
>Action [](start = 41, length = 6)

Still no suppressions data? Seems like we should have this, since it is a concept we exercise in the pattern matcher. Fine to defer.

Reply:
Added Suppressions and came up with a model to add arbitrary Properties. The other collections looked much more complex to decide how to serialize.